nflow-engine/src/main/java/io/nflow/engine/workflow/curated/MaintenanceWorkflowStarter.java

```diff
@@ -0,0 +1,46 @@
+package io.nflow.engine.workflow.curated;
+
+import io.nflow.engine.service.MaintenanceConfiguration;
+import io.nflow.engine.service.WorkflowInstanceService;
+import io.nflow.engine.workflow.instance.WorkflowInstance;
+import org.springframework.context.event.ContextStartedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import static org.joda.time.Period.days;
+import static org.joda.time.Period.years;
+
+@Component
+public class MaintenanceWorkflowStarter {
+  protected final WorkflowInstanceService instanceService;
+  protected final boolean insertOnStartup;
+
+  public MaintenanceWorkflowStarter(Environment env, WorkflowInstanceService instanceService) {
+    this.instanceService = instanceService;
+    this.insertOnStartup = env.getProperty("nflow.maintenance.start", Boolean.class, false);
+  }
+
+  protected String getDefaultCronSchedule() {
+    return "4 4 4 * * *";
+  }
+
+  protected MaintenanceConfiguration getDefaultConfiguration() {
+    return new MaintenanceConfiguration.Builder()
+            .withDeleteArchivedWorkflows().setOlderThanPeriod(years(1)).done()
+            .withArchiveWorkflows().setOlderThanPeriod(days(2)).done()
```

Comment:
This is pretty aggressive. Maybe a week or even month would be enough in most cases? At least in my project I use 90 days for archiving.

Reply:
What would be the harm in archiving aggressvively?

Reply:
Makes investigating stuff a bit more difficult.

Reply:
Seehttps://github.com/NitorCreations/nflow/pull/381

Reply:
Does not really help when you need to investigate stuff directly from db.

Reply:
Directly in db: it is as easy to search from archives than from main tables.
But I can later do a PR that provides a view that combines them

Reply:
Not really if the archive tables are huge and production tables are not. And this is not the only reason anyway. I will anyway need to override the default to much longer period in one of my projects, there's no need for more workarounds. I just think less aggressive defaults would be more safe. If the performance becomes a problem somewhere, it is super easy to configure more aggressive cleanup. Other way around, the damage has already been done.